### PR TITLE
Clean up 2.0.0 BuildTools PR

### PR DESCRIPTION
**Pull request by wtgodbe**
Addresses https://github.com/dotnet/coreclr/issues/11665

Also adds src/.nuget/optdata back to the repo, which I accidentally removed without replacing in my initial change.

CC @weshaggard @gkhanna79

### CODE REVIEW

**Comment by gkhanna79** [comment]
@wtgodbe Everything is failing - can you PTAL?

Also, can you please elaborate what cleanup is this about? The issue does not have any details.

**Comment by wtgodbe** [comment]
I just updated the issue, which now reads:

> I need to address feedback from #11620, which was merged before general cleanup changes could be made/tested

Will take a look at the failures now

**Comment by wtgodbe** [comment]
@weshaggard non-Windows jobs are failing to restore the optdata package with the following error:

> 12:21:19 /Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/Tools/dotnetcli/sdk/2.0.0-preview1-005724/NuGet.targets(97,5): error : Failed to retrieve information about 'optimization.PGO.CoreCLR' from remote source '/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/src/.nuget/optdata/https:/dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json'. [/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/src/.nuget/optdata/optdata.csproj] [/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/build.proj]
12:21:19 
/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/build.proj(32,5): error MSB3073: The command "/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/Tools/dotnetcli/dotnet restore --packages "/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/packages/" "/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/src/.nuget/optdata/optdata.csproj" --source "https:/dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json"" exited with code 1.

'https:/dotnet.myget.org...' looks wrong, but this must have been happening before (That comes from https://github.com/dotnet/coreclr/blob/master/build.proj#L29, which I didn't change - guessing the '//' gets treated as an escape sequence for '/'. I tried locally, and was able to repro with 1 '/', but '//' gave:

> 404 (Feed does not exist)

Any idea what's happening here?

**Comment by wtgodbe** [comment]
> source '/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/src/.nuget/optdata/https:/dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json'.

Seems suspect as well

**Comment by wtgodbe** [comment]
> source '/Users/dotnet-bot/j/workspace/dotnet_coreclr/master/checked_osx10.12_prtest/src/.nuget/optdata/https:/dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json'.

Seems suspect as well

**Comment by gkhanna79** [comment]
Is the (lack of) colon encoding the issue?

**Comment by wtgodbe** [comment]
What do you mean by colon encoding?

File: dir.props [hunk 1]
```diff
@@ -92,32 +92,10 @@
...
-
-  <PropertyGroup>
-    <DotnetToolCommand Condition="'$(DotnetToolCommand)'=='' and '$(OsEnvironment)'!='Unix'">$(DotnetCliPath)dotnet.exe</DotnetToolCommand>
-    <DotnetToolCommand Condition="'$(DotnetToolCommand)'=='' and '$(OsEnvironment)'=='Unix'">$(DotnetCliPath)dotnet</DotnetToolCommand>
-
-    <DnuRestoreCommand>"$(DotnetToolCommand)"</DnuRestoreCommand>
-    <DnuRestoreCommand>$(DnuRestoreCommand) restore</DnuRestoreCommand>
-    <DnuRestoreCommand Condition="'$(OsEnvironment)'=='Windows_NT'">$(DnuRestoreCommand) --packages "$(PackagesDir.TrimEnd('\\'))"</DnuRestoreCommand>
-    <DnuRestoreCommand Condition="'$(OsEnvironment)'!='Windows_NT'">$(DnuRestoreCommand) --packages "$(PackagesDir.TrimEnd('\'))"</DnuRestoreCommand>
+    <DotnetToolArgs>restore</DotnetToolArgs>
```
Comment (weshaggard) [inline dir.props]: Why can't the commands use the DotnetRestoreCommand?

File: tests/dir.props [hunk 2]
```diff
@@ -64,50 +64,23 @@
...
-    <DotnetToolCommand Condition="'$(DotnetToolCommand)'=='' and '$(OsEnvironment)'!='Unix'">$(DotnetCliPath)dotnet.exe</DotnetToolCommand>    
-    <DotnetToolCommand Condition="'$(DotnetToolCommand)'=='' and '$(OsEnvironment)'=='Unix'">$(DotnetCliPath)dotnet</DotnetToolCommand>    
-
-    <DnuRestoreSource>@(DnuSourceList -> '--source %(Identity)', ' ')</DnuRestoreSource>
-
-    <DnuRestoreCommand>"$(DotnetToolCommand)"</DnuRestoreCommand>
-    <DnuRestoreCommand>$(DnuRestoreCommand) restore</DnuRestoreCommand>
-    <DnuRestoreCommand>$(DnuRestoreCommand) --packages "$(PackagesDir.TrimEnd('/\'.ToCharArray()))" $(DnuRestoreSource)</DnuRestoreCommand>
+  <PropertyGroup>  
+    <DotnetRestoreSource>@(DotnetSourceList -> '--source %(Identity)', ' ')</DotnetRestoreSource>
```
Comment (weshaggard) [inline tests/dir.props]: Can you not use DotnetRestoreCommand here?

File: build.proj [hunk 3]
```diff
@@ -25,11 +25,11 @@
   </Target>
 
   <PropertyGroup>
-    <OptDataProjectJson>$(SourceDir).nuget/optdata/project.json</OptDataProjectJson>
+    <OptDataCsproj>$(SourceDir).nuget/optdata/optdata.csproj</OptDataCsproj>
     <OptDataPackageFeed>https://dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json</OptDataPackageFeed>
   </PropertyGroup>
   <Target Name="RestoreOptData">
-    <Exec Condition="Exists('$(OptDataProjectJson)')" Command="$(DnuRestoreCommand) &quot;$(OptDataProjectJson)&quot; --source &quot;$(OptDataPackageFeed)&quot;" />
+    <Exec Condition="Exists('$(OptDataCsproj)')" Command="$(DotnetToolCommand) $(DotnetToolArgs) &quot;$(OptDataCsproj)&quot; --source &quot;$(OptDataPackageFeed)&quot;" />
```
Comment (weshaggard) [inline build.proj]: The old DnuRestoreCommand include other feeds as well, I think that is still needed. Switching this to DotnetRestoreCommand should fix the problem.

**Comment by weshaggard** [comment]
https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/tests/helixperftasks.targets#L56

There are some other places you might need to update. Make sure you do a search for DNU and clear those up.

**Comment by wtgodbe** [comment]
I guess I missed that one. I did a search for "DnuRestore", since "dnu" gave too many results (builDNUmber, etc). Searching for 'DnuSource' now.

**Comment by wtgodbe** [comment]
@weshaggard after updating to use DotnetRestoreCommand, I'm still seeing the same failure as above in non-windows runs: https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_osx10.12_prtest/3009/console

File: build.proj [hunk 4]
```diff
@@ -25,11 +25,11 @@
   </Target>
 
   <PropertyGroup>
-    <OptDataProjectJson>$(SourceDir).nuget/optdata/project.json</OptDataProjectJson>
+    <OptDataCsproj>$(SourceDir).nuget/optdata/optdata.csproj</OptDataCsproj>
     <OptDataPackageFeed>https://dotnet.myget.org/F/dotnet-core-optimization-data/api/v3/index.json</OptDataPackageFeed>
   </PropertyGroup>
   <Target Name="RestoreOptData">
-    <Exec Condition="Exists('$(OptDataProjectJson)')" Command="$(DnuRestoreCommand) &quot;$(OptDataProjectJson)&quot; --source &quot;$(OptDataPackageFeed)&quot;" />
```
Comment (weshaggard) [inline build.proj]: Looks like the old project.json file doesn't even exist so I suspect this restore hasn't worked for a long time.

Comment (weshaggard) [inline build.proj]: So we should just disable this and have @dpodder look at it.

**Comment by wtgodbe** [comment]
@dotnet-bot test Windows_NT x86 Checked Build and Test please

**Comment by gkhanna79** [comment]
Please prepare or for 2.0.0 as well.

**Pull request merged**